neuraxle/data_container.py

```diff
@@ -142,44 +145,90 @@ def hash_summary(self):
             m.update(str.encode(str(current_id)))
         return m.hexdigest()
 
-    def convolved_1d(self, stride, kernel_size) -> Iterable['DataContainer']:
+    def batch(
+            self,
+            batch_size: int,
+            include_incomplete_batch: bool = False,
+            default_value_data_inputs=None,
+            default_value_expected_outputs=None
+    ) -> Iterable['DataContainer']:
         """
         Returns an iterator that iterates through batches of the DataContainer.
 
-        :param stride: step size for the convolution operation
-        :param kernel_size:
+
+        .. code-block:: python
+
+            data_container = DataContainer(data_inputs=np.array(list(range(10)))
+            for data_container_batch in data_container.batch(batch_size=2):
+                print(data_container_batch.data_inputs)
+                print(data_container_batch.expected_outputs)
+            # [array([0, 1]), array([2, 3]), ..., array([8, 9])]
+
+            data_container = DataContainer(data_inputs=np.array(list(range(10)))
+            for data_container_batch in data_container.batch(batch_size=3, include_incomplete_batch=False):
+                print(data_container_batch.data_inputs)
+            # [array([0, 1, 2]), array([3, 4, 5]), array([6, 7, 8])]
+
+            data_container = DataContainer(data_inputs=np.array(list(range(10)))
+            for data_container_batch in data_container.batch(
+                batch_size=3,
+                include_incomplete_batch=True,
+                default_value_data_inputs=None,
+                default_value_expected_outputs=None
+            ):
+                print(data_container_batch.data_inputs)
+            # [array([0, 1, 2]), array([3, 4, 5]), array([6, 7, 8]), array([9, None, None])]
+
+            data_container = DataContainer(data_inputs=np.array(list(range(10)))
+            for data_container_batch in data_container.batch(
+                batch_size=3,
+                include_incomplete_batch=True,
+                default_value_data_inputs=AbsentValuesNullObject()
+            ):
+                print(data_container_batch.data_inputs)
+            # [array([0, 1, 2]), array([3, 4, 5]), array([6, 7, 8]), array([9])]
+
+
+        :param batch_size: number of elements to combine into a single batch
+        :param include_incomplete_batch: (Optional.) A bool representing
+        whether the last batch should be dropped in the case it has fewer than
+        `batch_size` elements; the default behavior is not to drop the smaller
+        batch.
+        :param default_value_data_inputs: expected_outputs default fill value
+        for padding and values outside iteration range, or absent value null object
+        to trim absent values from the batch
```

Comment:
Please document this so as to link and point to the `AbsentValuesNullObject` class. Also add a `seealso` please. I would document also the MiniBatchSequentialPipeline the same way with some see alsos and mentionning the role of the AbsentValuesNullObject.